coderd/prometheusmetrics/aggregator.go

```diff
@@ -102,13 +108,28 @@ func hashKey(req *updateRequest, m *agentproto.Stats_Metric) metricKey {
 var _ prometheus.Collector = new(MetricsAggregator)
 
 func (am *annotatedMetric) asPrometheus() (prometheus.Metric, error) {
-	labels := make([]string, 0, len(agentMetricsLabels)+len(am.Labels))
-	labelValues := make([]string, 0, len(agentMetricsLabels)+len(am.Labels))
+	var (
+		baseLabelNames  = am.aggregateByLabels
+		baseLabelValues []string
+		extraLabels     = am.Labels
+	)
+
+	for _, label := range am.aggregateByLabels {
```

Comment:
Perhaps this was the intent here?

```suggestion
	for _, label := range baseLabelNames {
```